src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -107,6 +110,56 @@ public async Task<IActionResult> GetHostStatus([FromServices] IScriptHostManager
             return Ok(status);
         }
 
+        /// <summary>
+        ///  Currently, anyone in Reader role can access this information.
+        ///  If this API is extended to include any secrets, it will need to be
+        ///  locked down to only Contributor roles.
+        /// </summary>
+        [HttpGet]
+        [Route("admin/host/workerProcesses")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> GetWorkerProcesses([FromServices] IScriptHostManager scriptHostManager)
+        {
+            if (!Utility.TryGetHostService(scriptHostManager, out IWebHostRpcWorkerChannelManager webHostLanguageWorkerChannelManager))
+            {
+                return StatusCode(StatusCodes.Status503ServiceUnavailable);
+            }
+
+            var processes = new List<WorkerProcessInfo>();
+            string workerRuntime = _environment.GetFunctionsWorkerRuntime();
+
+            var webhostChannelDictionary = webHostLanguageWorkerChannelManager.GetChannels(workerRuntime);
+
+            List<IRpcWorkerChannel> channels = new List<IRpcWorkerChannel>();
+            if (webhostChannelDictionary is not null)
+            {
+                foreach (var pair in webhostChannelDictionary)
+                {
+                    var workerChannel = await pair.Value.Task;
+                    channels.Add(workerChannel);
+                }
+            }
+
+            if (Utility.TryGetHostService(scriptHostManager, out IJobHostRpcWorkerChannelManager jobHostLanguageWorkerChannelManager))
+            {
+                channels.AddRange(jobHostLanguageWorkerChannelManager.GetChannels(workerRuntime));
+            }
+
+            foreach (var channel in channels)
+            {
+                var processInfo = new WorkerProcessInfo()
+                {
+                    ProcessId = channel.WorkerProcess.Process.Id,
+                    ExecutableName = channel.WorkerProcess.Process.ProcessName,
+                    DebugEngine = Utility.GetDebugEngineInfo(channel.WorkerConfig, workerRuntime),
+                    IsEligibleForOpenInBrowser = false
```

Comment:
Confused about this property. How this will be used when value of this prop is true/false ?

Reply:
This property is required by the VS, it is so far false for all languages.

Reply:
I think there is a different type of debugger attachment action that would happen if it is true.

Reply:
Got it. Thanks!